ci/appveyor-cpp-setup.bat

```diff
@@ -55,6 +55,11 @@ mamba update -q -y -c conda-forge --all || exit /B
 @rem Create conda environment
 @rem
 
+@rem Workaround for ARROW-17172
+@rem Due to failing test_cython_api() import of an external module in the
+@rem subprocess this env var is added for os.add_dll_directory to work as
+@rem expected in the conda environment and for the import to succeed.
+@rem See: https://github.com/ContinuumIO/anaconda-issues/issues/12475
```

Comment:
We don't call `add_dll_directory` so I'm not sure that's the right explanation. I would stay vague about it, for example:
```suggestion
@rem This seems necessary for test_cython.py to succeed, otherwise
@rem the extension module being built would fail loading in a subprocess.
```